File: 链表/24.两两交换链表中的节点.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
package main

type ListNode struct {
	Val  int
	Next *ListNode
}

func swapPairs(head *ListNode) *ListNode {
	res := &ListNode{-1, head}
	pre := res
	for pre.Next != nil && pre.Next.Next != nil {
		tmp := pre.Next.Next.Next
		tmp2 := pre.Next.Next
		tmp2.Next = pre.Next
		pre.Next.Next = tmp
		pre.Next = tmp2
		pre = pre.Next.Next
	}
	return res.Next
}

File: 链表/24.两两交换链表中的节点.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        ListNode res = new ListNode(-1,head);
        ListNode pre = res;
        while(pre.next!=null && pre.next.next!=null){
            ListNode tmp = pre.next.next.next;
            ListNode tmp2 = pre.next.next;
            pre.next.next.next=pre.next;
            pre.next.next=tmp;
            pre.next=tmp2;
            pre=pre.next.next;
        }
        return res.next;
    }
}